### PR TITLE
Move redeemTickets logic to redux

### LINKED ISSUE #164: Node: move handleRedeemAllTickets logic to Redux
Definition of done:
```

  const handleRedeemAllTickets = () => {
    set_redeeming(true);
    dispatch(
      actionsAsync.redeemTicketsThunk({
        apiEndpoint: loginData.apiEndpoint!,
        apiToken: loginData.apiToken!,
      }),
    )
      .unwrap()
      .then(() => {
        set_redeeming(false);
        set_redeemSuccess(true);
        handleRefresh();
      })
      .catch((e) => {
        set_redeeming(false);
        set_redeemSuccess(false);
        set_redeemErrors([
          ...redeemErrors,
          {
            error: e.error,
            status: e.status,
          },
        ]);
      });
  };
```
 - similar to above should be moved to Redux
 - Redux should know if redeming is in progress (boolean in Redux instead of  `const [redeeming, set_redeeming] = useState(false);`) 
 - redeemErrors should be also in Redux
 - connect ```
             <IconButton
              iconComponent={<ExitToAppIcon/>}
              tooltipText='Redeem all tickets'
              reloading={redeeming}
              onClick={handleRedeemAllTickets}
            />``` to redux redeeming state

### PR DESCRIPTION
**Pull request by ItsLuized**
Closes #164 

### Overview

This pull request aims to provide the functionality of handling when redeemTicketsThunk is loading and when it has errors in a global manner to reduce the amount of repeated amount of code.

The new redeemTickets state structure is:
```
redeemTickets: {
    isFetching: false,
    error: {
      error: undefined,
      status: undefined,
    },
  },
```

### CODE REVIEW

File: src/store/slices/node/initialState.ts [hunk 1]
```diff
@@ -98,7 +98,13 @@ type InitialState = {
   messagesWebsocketStatus: WebsocketConnectionStatus;
   logsWebsocketStatus: WebsocketConnectionStatus;
   closeChannel: { isFetching: boolean };
-  redeemTickets: { isFetching: boolean };
+  redeemTickets: {
+    isFetching: boolean;
+    error: {
+      status: string | undefined;
+      error: string | undefined;
+    };
```
Comment (diegoalzate) [inline src/store/slices/node/initialState.ts:106]: i think we can just have error: string

what do you think @mjadach-iv ?

**Review by diegoalzate — CHANGES_REQUESTED** [review]

File: src/store/slices/node/initialState.ts (hunk 1 above)
Comment (mjadach-iv) [reply src/store/slices/node/initialState.ts:106]: yeah, easier - better

**Review by mjadach-iv — CHANGES_REQUESTED** [review]

File: src/sections/node/tickets.tsx [hunk 2]
```diff
@@ -44,28 +44,12 @@ function TicketsPage() {
...
     dispatch(
       actionsAsync.redeemTicketsThunk({
         apiEndpoint: loginData.apiEndpoint!,
         apiToken: loginData.apiToken!,
       }),
-    )
-      .unwrap()
-      .then(() => {
-        set_redeeming(false);
-        handleRefresh();
```
Comment (diegoalzate) [inline src/sections/node/tickets.tsx:57]: noticed that handleRefresh is no longer going to be called. Maybe not needed let me know.

**Review by diegoalzate — COMMENTED** [review]
small question above

File: src/sections/node/tickets.tsx (hunk 2 above)
Comment (ItsLuized) [reply src/sections/node/tickets.tsx:57]: It is necessary, nice catch. I will move the handle refresh to happen after the reeemTickets

File: src/sections/node/tickets.tsx [hunk 3]
```diff
@@ -44,28 +44,13 @@ function TicketsPage() {
...
-        set_redeemErrors([
-          ...redeemErrors,
-          {
-            error: e.error,
-            status: e.status,
-          },
-        ]);
-      });
+    );
+    handleRefresh();
```
Comment (diegoalzate) [inline src/sections/node/tickets.tsx:53]: this will refresh before the action is finished. do .unwrap and then .then and inside that then you can refresh.

**Review by diegoalzate — CHANGES_REQUESTED** [review]

**Pull request merged**